### DISCUSSION TITLE
Timepicker instead of hours to show for map history

### DISCUSSION THREAD

**Discussion by eifinger** [home-assistant/frontend#11253] [Other feature requests]
## The request

I was thinking on how to replace the Google Maps Timeline with Homeassistant data for a long time. So I was absolutely thrilled when the Map History got released thanks to @Hoytron .

I only wish I could choose the `hours_to_show` on demand. Switching between `last_hour`, `last_day`, `last_week` and a custom timeframe.

## Additional information

I didn't have time (yet) to look deeper into the frontend code whether this request is feasible at all and if yes how it could be implemented. I will add ideas to this issue when I come up with them. Just wanted to get the idea out there. 😃

**Comment by Hoytron** [comment]
Hi @eifinger , I‘m happy you like the new feature! 
So if I get the idea right your idea is a group of buttons on the map card, to toggle different time spans? 
A Off/Zero button is maybe helpful too.

The convenience of these buttons is obvious and the code is probably simple. 
But I am not sure which preset of buttons fits in the concept of HA. 
How long home assistant keeps the data is configurable, so we should discuss suitable presets.

**Comment by eifinger** [comment]
That's exactly what I meant.
